src/cml.js

```diff
@@ -282,23 +293,24 @@ class CML {
       if (url) return render_pr(url);
     } else {
       try {
-        await exec(`git config --local user.email "david@iterative.ai"`);
-        await exec(`git config --local user.name "cml-bot"`);
-        await exec('git config advice.addIgnoredFile false');
-
-        if (this.driver !== 'github') {
-          const repo = new URL(this.repo);
-          repo.password = this.token;
-          repo.username = driver.user_name;
-
-          await exec(`git remote rm origin`);
-          await exec(`git remote add origin "${repo.toString()}.git"`);
+        await exec(`git config --local user.email "${user_email}"`);
+        await exec(`git config --local user.name "${user_name}"`);
+
+        if (isCI()) {
+          if (this.driver === 'gitlab') {
+            const repo = new URL(this.repo);
+            repo.password = this.token;
+            repo.username = driver.user_name;
+
+            await exec(`git remote rm ${remote}`);
+            await exec(`git remote add ${remote} "${repo.toString()}.git"`);
+          }
```

Comment:
```suggestion
        if (isCI() && this.driver === 'gitlab') {
          const repo = new URL(this.repo);
          repo.password = this.token;
          repo.username = driver.user_name;

          await exec(`git remote rm ${remote}`);
          await exec(`git remote add ${remote} "${repo.toString()}.git"`);
```

Reply:
It's written that way because Im not happy either with that. It should be managed by the driver but probably shares the code with all the rest (GH, BB) however in GH and BB is not needed to do it... I'm not dully happy with that...

Reply:
resolving in favour of #529